src/foam/dao/SequenceNumberDAO.java

```diff
@@ -42,21 +52,23 @@ private PropertyInfo getProperty_() {
    * Calculates the next largest value in the sequence
    */
   private void calcDelegateMax_() {
-    Max maxSink = (Max) new Max().setArg1(getProperty_());
-    ((AbstractDAO) getDelegate()).select(maxSink);
-    int maxValue = (int) maxSink.getValue();
-    if (maxValue < 1) return;
-    this.setValue(maxValue + 1);
+    Sink sink = MLang.MAX(getProperty_());
+    getDelegate().select(sink);
+    this.setValue((long) (((Max) sink).getValue() + 1.0));
+  }
+
+  public FObject put(FObject obj) {
+    return put_(getX(), obj);
   }
 
   public FObject put_(X x, FObject obj) {
     calcDelegateMax_();
```

Comment:
This is going to be very slow.  You should only calcDelegateMax on the first put(), not on every put.